src/audio/codec_adapter/codec_adapter.c

```diff
@@ -71,12 +72,91 @@ static struct comp_dev *codec_adapter_new(const struct comp_driver *drv,
 	}
 
 	comp_set_drvdata(dev, cd);
+	/* Copy setup config */
+	ret = load_setup_config(dev, ipc_codec_adapter->data, ipc_codec_adapter->size);
+	if (ret) {
+		comp_err(dev, "codec_adapter_new() error %d: config loading has failed.",
+			 ret);
+		goto err;
+	}
 
 	dev->state = COMP_STATE_READY;
 	cd->state = PP_STATE_CREATED;
 
 	comp_cl_info(&comp_codec_adapter, "codec_adapter_new() done");
 	return dev;
+err:
+	rfree(cd);
+	rfree(dev);
+	return NULL;
+}
+
+static inline int validate_setup_config(struct ca_config *cfg)
+{
+	/* TODO: validate codec_adapter setup parameters */
+	return 0;
+}
+
+/**
+ * \brief Load setup config for both codec adapter and codec library.
+ * \param[in] dev - codec adapter component device pointer.
+ * \param[in] cfg - pointer to the configuration data.
+ * \param[in] size - size of config.
+ *
+ * The setup config comprises of two parts - one contains essential data
+ * for the initialization of codec_adapter and follows struct ca_config.
+ * Second contains codec specific data needed to setup the codec itself.
+ * The leter is send in a TLV format organized by struct codec_param.
+ *
```

Comment:
I assume you want instead of ``The leter is send`` -> ``The latter is sent``? Just to avoid confusion.

Reply:
yep, thanks for spotting.